### PR TITLE
#163288966 Change id types to UUIDV4

### PR DESCRIPTION
**Pull request by Bobsar0**
#### What does this PR do?
Change all model id's to type UUID

#### Description of Task to be completed?
Change all model id's from INTEGER to type UUID

#### How should this be manually tested?
 • clone the repo
 • cd into the repo
 • [Setup development database](https://github.com/andela/neon-ah-backend/pull/14#issue-244353384)
 • run `npm install`
 • run `npm run migrate && npm run seed`
 • Check the id columns of the seeded users, roles and authTypes tables.

#### What are the relevant pivotal tracker stories?
[Change id types to UUID](https://www.pivotaltracker.com/story/show/163288966)

### CODE REVIEW

File: server/migrations/20190111125359-create-role.js [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
-      defaultValue: Sequelize.literal('CURRENT_TIMESTAMP')
-    },
-    updatedAt: {
-      allowNull: false,
-      type: Sequelize.DATE,
-      defaultValue: Sequelize.literal('CURRENT_TIMESTAMP')
-    }
-  }),
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.query('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
```
Comment (jesseinit) [inline server/migrations/20190111125359-create-role.js:3]: What's happening on this Line?

File: server/migrations/20190111125231-create-user.js [hunk 2]
```diff
@@ -1,62 +1,67 @@
...
+            type: Sequelize.STRING
+          },
+          notifySettings: {
+            type: Sequelize.BOOLEAN,
+            defaultValue: true
+          },
+          roleId: {
+            type: Sequelize.UUID,
+            allowNull: false,
+            defaultValue: '3ceb546e-054d-4c1d-8860-e27c209d4ae3',
```
Comment (jesseinit) [inline server/migrations/20190111125231-create-user.js:47]: Can you color on why this is hardcoded?

Comment (jesseinit) [reply server/migrations/20190111125231-create-user.js:47]: Ohhh...I get now.
Nice work.

**Review by jesseinit — APPROVED** [review]
With the push is building, the test passing.
This looks pretty good to me.

File: server/migrations/20190111125231-create-user.js (hunk 2 above)
Comment (Makwe-O) [reply server/migrations/20190111125231-create-user.js:47]: LGTM

**Comment by Makwe-O** [comment]
LGTM

File: server/migrations/20190111125359-create-role.js (hunk 1 above)
Comment (rafmme) [reply server/migrations/20190111125359-create-role.js:3]: LGTM, yet. Don't forget to close all open conversations when resolved.

Comment (Bobsar0) [reply server/migrations/20190111125359-create-role.js:3]: @jesseinit the extension is used to autogenerate uuids upon row insertion

**Review by seunkoko — APPROVED** [review]

**Pull request merged**